### PR TITLE
Fv 366 darstellung langer straßennamen im blp

### PR DESCRIPTION
**Pull request by sfi2022**
# Description

Darstellung langer Strassennamen mit verschiedenen Trennern ("-", " ") korrigiert. Code und Darstellungsoptionen refactored.
Releasenotes Sprint 11 muss noch komplementiert werden, wenn er existiert.

# Issue References

FV-366
 
# Definition of Done

- [x] Acceptance criteria are met
- [x] Testing is done (unit-tests, DEV-environment)
- [x] Build/Test workflow has successfully finished
- [x] Release notes are complemented
- [x] Documentation is complemented (operator manual, system specification, etc.)

## Summary by CodeRabbit

* **New Features**
  * Unified street-name formatter producing balanced one- or two-line labels across charts.

* **Style**
  * Adjusted text direction/alignment for select node labels to improve readability.
  * Minor SVG text/legend style tweaks for more consistent rendering.

* **Bug Fixes**
  * Fixed inconsistent wrapping/alignment that caused misrendered or truncated labels.

* **Tests**
  * Updated and expanded tests to validate new formatting and edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/it-at-m/dave-frontend/pull/615?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: frontend/src/util/StrassennameUtils.ts [hunk 1]
```diff
@@ -1,51 +1,196 @@
...
+   * Liefert die beiden Zeilen (first/second) direkt für einen Knotenarm.
+   * Intern ruft diese Funktion die Tokenisierung (getStreetname / Segmenter) und dann
+   * balanceLines auf.
+   */
+  function getStreetLines(
+    knotenarm: LadeKnotenarmDTO | undefined,
+    maxChars?: number
+  ): Array<string> {
+    const pieces = getStreetname(knotenarm, maxChars);
+    return balanceLines(pieces, getMostFrequentSeparator(pieces));
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply frontend/src/util/StrassennameUtils.ts]: The 'Easter egg' was astutely spotted. However, the risk is being taken. The names are entered manually. Consequently, you have to correct them there if they don't suit you.

File: frontend/src/util/StrassennameUtils.ts [hunk 2]
```diff
@@ -1,51 +1,196 @@
...
+    }
+
+    // 1) Fallback: An Leerstelle trennen falls vorhanden
+    const lastSpaceBefore = strasse.lastIndexOf(" ");
+    if (lastSpaceBefore > 0) {
+      return [
+        strasse.substring(0, lastSpaceBefore + 1).trim(),
+        strasse.substring(lastSpaceBefore + 1).trim(),
+      ];
+    }
```
[the remark this answers is not in the document]
Comment (sfi2022) [reply frontend/src/util/StrassennameUtils.ts]: Accepted. I corrected this.

File: frontend/src/components/zaehlstelle/optionsmenue/panels/verkehrsbeziehungen/QuerschnittJeStrassenseiteForm.vue [hunk 3]
```diff
@@ -588,7 +588,9 @@ function resetForm(): void {
 }
 
 function prepareStreetnames(): void {
-  firstStreetname.value = streetnameUtils.getStreetname(firstKnotenarm.value);
-  secondStreetname.value = streetnameUtils.getStreetname(secondKnotenarm.value);
+  firstStreetname.value = strassennameUtils.getStreetLines(firstKnotenarm.value);
```
Comment (MarinaSchm) [inline frontend/src/components/zaehlstelle/optionsmenue/panels/verkehrsbeziehungen/QuerschnittJeStrassenseiteForm.vue]: Formatierung: run fix:prettier

File: frontend/src/util/StrassennameUtils.ts [hunk 4]
```diff
@@ -1,51 +1,196 @@
 import type LadeKnotenarmDTO from "@/types/zaehlung/LadeKnotenarmDTO";
 
-export function useStreetname() {
+export function useStrassennameUtils() {
   /**
    * Extrahiert und splittet den Straßennamen eines Knotenarms
    * @param knotenarm optionaler Knotenarm DTO
+   * @param maxChars maximale Zeichenanzahl in der ersten Zeile (Default: 20)
```
Comment (MarinaSchm) [inline frontend/src/util/StrassennameUtils.ts]: Kommentar anpassen: Default 17

File: frontend/src/components/zaehlstelle/optionsmenue/panels/verkehrsbeziehungen/QuerschnittJeStrassenseiteForm.vue (hunk 3 above)
Comment (sfi2022) [reply frontend/src/components/zaehlstelle/optionsmenue/panels/verkehrsbeziehungen/QuerschnittJeStrassenseiteForm.vue]: Done.

File: frontend/src/util/StrassennameUtils.ts (hunk 4 above)
Comment (sfi2022) [reply frontend/src/util/StrassennameUtils.ts]: Done

File: frontend/src/util/StrassennameUtils.ts [hunk 5]
```diff
@@ -1,51 +1,196 @@
...
+    }
+
+    // 5) Letzte Rettung: mittig schneiden (mit optionaler Silbentrennung würde das verbessert)
+    const mid = Math.ceil(strasse.length / 2);
+    return [
+      strasse.substring(0, mid).trim() + "-",
+      strasse.substring(mid).trim(),
+    ];
+  }
+
```
Comment (MarinaSchm) [inline frontend/src/util/StrassennameUtils.ts:115]: Hier wäre noch ein kurzer Javadoc schön um auf einen Blick zu sehen, was die Funktion macht. Den return Wert  in der Deklaration noch angeben.

File: frontend/src/util/StrassennameUtils.ts [hunk 6]
```diff
@@ -1,51 +1,196 @@
...
+    // Füge den Trenner am Ende der ersten Zeile hinzu, falls er im Input vorhanden war
+    if (firstLineEndsWithSeparator) {
+      firstLineResult += separatorAtEnd;
+    }
+
+    return [firstLineResult, secondLineResult];
+  }
+
+  function getMostFrequentSeparator(array: string[]) {
```
Comment (MarinaSchm) [inline frontend/src/util/StrassennameUtils.ts]: Hier fehlt auch die Angabe des return-Werts (und Javadoc).

File: frontend/src/util/StrassennameUtils.ts (hunk 5 above)
Comment (MarinaSchm) [reply frontend/src/util/StrassennameUtils.ts:115]: Zeile 115 meine ich: function balanceLines

File: frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue [hunk 7]
```diff
@@ -5907,28 +5909,28 @@ watch(
     () => optionen.value.chosenLaengsverkehre,
   ],
   async () => {
-    streetnameNodeOne.value = fjs.getStreetname(
```
Comment (MarinaSchm) [inline frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue:5909]: Der Code ab hier ist das selbe wie ab in Zeile 5875-5899. Ich würde das in eine Helper-Funktion auslagern.

File: frontend/src/components/zaehlstelle/charts/BelastungsplanQuSvg.vue [hunk 8]
```diff
@@ -3918,28 +3918,28 @@ watch(
     () => activeZaehlung.value.knotenarme,
   ],
   async () => {
-    streetnamesNode1.value = streetnameUtils.getStreetname(
```
Comment (MarinaSchm) [inline frontend/src/components/zaehlstelle/charts/BelastungsplanQuSvg.vue:3921]: Hier auch doppelter Code wie in onMounted. Würde ich in eine Helper-Funktion auslagern.

File: frontend/src/util/StrassennameUtils.ts (hunk 5 above)
Comment (sfi2022) [reply frontend/src/util/StrassennameUtils.ts:115]: Done

File: frontend/src/util/StrassennameUtils.ts (hunk 6 above)
Comment (sfi2022) [reply frontend/src/util/StrassennameUtils.ts]: Done

File: frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue (hunk 7 above)
Comment (sfi2022) [reply frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue:5909]: Ich habe das Refactoring jetzt mal gemacht. Ist aber eigentlich nicht Teil des vorliegenden Tickets.
Ich habe den BelastungsplanFjsSVG nur angefasst um die Code Änderungen FjsUtil nach StrassennamenUtil anzupassen.

File: frontend/src/components/zaehlstelle/charts/BelastungsplanQuSvg.vue (hunk 8 above)
Comment (sfi2022) [reply frontend/src/components/zaehlstelle/charts/BelastungsplanQuSvg.vue:3921]: Wie oben.

File: frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue (hunk 7 above)
Comment (MarinaSchm) [reply frontend/src/components/zaehlstelle/charts/BelastungsplanFjsSvg.vue:5909]: Achso, ja hast eigentlich Recht. Aber danke fürs trotzdem machen :)

**Review by MarinaSchm — APPROVED** [review]
